tests/migration/patch_and_reboot_system.pm

```diff
@@ -6,23 +6,35 @@
 # Summary: Fully patch the system before conducting migration and then reboot
 # Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
 
-use base "consoletest";
+use base 'y2_installbase';
 use strict;
 use warnings;
 use testapi;
 use utils;
 use version_utils qw(is_desktop_installed is_upgrade);
+use power_action_utils 'power_action';
 use migration;
 
 sub run {
-    select_console 'root-console';
-
     my ($self) = @_;
 
-    fully_patch_system();
-    enter_cmd "reboot";
-    $self->wait_boot(textmode => !is_desktop_installed(), ready_time => 600, bootloader_time => 300);
     select_console 'root-console';
+
+    # Repeatedly call zypper patch until it returns something other than 103 (package manager updates)
+    my $ret = 1;
+    for (1 .. 3) {
+        $ret = zypper_call("patch --with-interactive -l", exitcode => [0, 4, 102, 103], timeout => 6000);
+        last if $ret != 103;
+    }
+    if (($ret == 4) && is_sle('>=12') && is_sle('<15')) {
+        my $para = '';
+        $para = '--force-resolution' if get_var('FORCE_DEPS');
+        $ret = zypper_call("patch --with-interactive -l $para", exitcode => [0, 102], timeout => 6000);
+        save_screenshot;
+    }
+    die "Zypper failed with $ret" if ($ret != 0 && $ret != 102);
+    power_action('reboot', keepconsole => 1);
```

Comment:
Before reboot we better to do sync to avoid broken of file system.

# Perform sync ahead of reboot to flush filesystem buffers
            assert_script_run 'sync', 600;